api/watchlist/watchlistRouter.js

```diff
@@ -0,0 +1,112 @@
+const express = require('express');
+const authRequired = require('../middleware/authRequired');
+const validateUserId = require('../middleware/validate-user-id');
+const Watchlist = require('./watchlist-bridgesModel');
+const { removeWatchlist, findWatchlist } = require('./watchlist-bridgesModel');
+const router = express.Router();
+
+// HTTP GET to retrieve the authenticated user's watchlist
+
+router.get('/:id', authRequired, function (req, res) {
+  const id = String(req.params.id);
+  Watchlist.findWatchlist(id)
+    .then((list) => {
+      if (list) {
+        res.status(200).json(list);
+      } else {
+        res.status(404).json({ error: 'Watchlist Not Found' });
+      }
+    })
+    .catch((err) => {
+      res.status(500).json({ error: err.message });
+    });
+});
+
+// HTTP POST to create a new watchlist for a user
+
+router.post('/:id', authRequired, validateUserId, async function (req, res) {
+  const notes = String(req.body.notes);
+  const user = String(req.params.id);
+  const locations = req.body.locations;
+  const title = String(req.body.title);
+  console.log(notes, user, locations, title);
+
+  try {
```

Comment:
is this being used?

Reply:
It was only for testing - it's been removed now.